### ISSUE TITLE
Handle denied permissions - White screen reported by App Store Connect

### ISSUE THREAD

**Issue by Bellacopia** [thunkable/thunkable-issues#550]
Last May 29th my app was rejected by App Store Connect for this reason, literally: 
“Specifically, your app failed to load content, showing a blank white page, after we answered to the location modal alert.”

![Fig](https://user-images.githubusercontent.com/63849566/83577912-c084cd80-a535-11ea-8859-74379db5a2f9.png)

During our internal tests it had never happened before and therefore it took a few days to be able to reproduce the error and here are our results, that is, the conditions in which the white screen appears and the app crashes. Note: This bug seems to occur regardless of the device (iPhone and iPad) and the iOS version (13.5 and 12.4.6).

On iOS, our app crashes to white screen:

CASE 1: If you answer negatively (“Don’t Allow”) to the location modal alert.
CASE 2: If you try to load the app after having switched off the “Location Services” in Settings/Privacy section (see figure below).

![Fig 2](https://user-images.githubusercontent.com/63849566/83578822-fd51c400-a537-11ea-8bc9-ed4be5b601a2.PNG)

At this point we decided to reproduce the error even on the Android version (which a few days ago has been accepted and now is available on Google Play). Here is the result:

On Android, our app crashes to white screen:

CASE 1: If you answer negatively (“Deny”) to the location modal alert (see figure below).
CASE 2: If you answer negatively  (“No, thanks”) to the alert that appears if switch off the “Location Access” in Settings/Location section (see figure below).

![Fig 3](https://user-images.githubusercontent.com/63849566/83578513-46554880-a537-11ea-8057-0195b703bf19.PNG)

![Fig 4](https://user-images.githubusercontent.com/63849566/83578871-122e5780-a538-11ea-809c-01d09911a098.PNG)

In conclusion, we can say that both versions (Android and iOS) compiled by Thunkable do not handle the event of negative response to the location modal alert.

Furthermore Thunkable iOS version does not even handle the event of “Location Services” switched off.

I would be very grateful to you if you would check our tests and fix these anomalies as soon as possible. It is not pleasant to have an app published on Google Play only.

Thanks a lot in advance

Ugo

**Comment by josmas** [comment]
Hi @Bellacopia thanks for the clear report; permission handling is always tricky.

I am going to slightly modify the title of this post before I move it, if you don't mind. Just to make the issue a bit clearer. Thanks!

**Comment by Bellacopia** [comment]
Thank you Jos,
when do you plan to look into this issue? I ask you because I would need an
estimate of the time it takes to resubmit the app to the App Store.
Thanks a lot in advance.
 Ugo

Il giorno ven 5 giu 2020 alle ore 19:14 Jos <notifications@github.com> ha
scritto:

**Comment by Bellacopia** [comment]
Good morning, more than ten days have passed since my first report of this problem and I find myself in SERIOUS DIFFICULTY and seriously embarrassed in managing the absence of our app from the App Store. But I have to tell you that the most annoying thing is the fact of feeling abandoned since you don't respond to my messages. As a PRO user I would have expected a different treatment. 
What I'm asking you is not an improvement or a new feature but the fixing of a serious ANOMALY that makes it impossible to publish on iOS. However, this task has NOT yet been assigned.
At least this time, I hope to hear from you soon.

**Comment by josmas** [comment]
Hi @Bellacopia unfortunately we do not have anything in place to handle rejected permissions. We had a meeting yesterday and we have a couple of ideas that we need to explore. We are going to try and return an error if a permission is not granted, instead of hanging the app, so the app can display a warning or handle it in any other way. We'll keep you posted if the solution is successful. Thanks!

**Comment by josmas** [comment]
Hi again @Bellacopia I was actually wrong when I said we have nothing to handle rejected Location permissions. Are you using the location component or some other component that needs the location permission?

With the Location sensor you can do this:

<img width="507" alt="Screen Shot 2020-06-10 at 12 16 40 AM" src="https://user-images.githubusercontent.com/194925/84210104-c9cee680-aaaf-11ea-90b3-472bf3171a5c.png">

And if the user denies the permission you will get this:
<img width="507" alt="Screen Shot 2020-06-10 at 12 17 19 AM" src="https://user-images.githubusercontent.com/194925/84210127-d5baa880-aaaf-11ea-8915-ea125655608c.png">

So basically if your user rejects, you can catch that error and show a warning saying that either the app does not work without Location, or that they will get limited functionality. I don't think we have this for all components that use Location, but you can use this component to make sure you have the permission you need. Thanks!

**Comment by Bellacopia** [comment]
Thank you very much, following your instructions I was able to solve the problem.

**Comment by josmas** [comment]
Great to hear @Bellacopia and apologies for not realising earlier that the error handling was already there. Thanks!
